### ISSUE TITLE
AVT 2 - Search Placeholder text in form input fields does not meet WCAG 2.0 color contrast requirements

### ISSUE THREAD

**Issue by snidersd** [carbon-design-system/carbon#3332]
## Environment
macOS Mojave version 10.14.5
Version 75.0.3770.100 (Official Build) (64-bit)
Carbon v10 - React
Voice Over

## Detailed Description
Use Color Contrast Analyzer to test the background color of the Search placeholder text against the background color of the text input field. See screenshot below:
<img width="795" alt="Screen Shot 2019-07-10 at 11 02 01 AM" src="https://user-images.githubusercontent.com/21676914/60985621-636fa000-a303-11e9-9fe6-0944b205a662.png">

See [WCAG Form Tutorial for placeholder text guidance](https://www.w3.org/WAI/tutorials/forms/instructions/#placeholder-text). 
Note: All form inputs with placeholder text appear to have this issue.

**Comment by tw15egan** [comment]
@aagonzales what's the deal with placeholder text?

**Comment by dakahn** [comment]
We should update this to be inline with our TextInput placeholder text.

**Comment by aagonzales** [comment]
My understanding is placeholder text doesn't have to pass contrast standards because it's not accessible in general because it disappears once the user starts typing. It should be using the token `$text-03`. If the search prompt needs to pass accessibility then it should use the token `$text-02`.

Using `$text-03` for placeholder text is what we do for TextInput

**Comment by snidersd** [comment]
Here is a link to the [WCAG Tutorial for Forms](https://www.w3.org/WAI/tutorials/forms/instructions/). It discusses placeholder accessibility in the style section. Also, the W3C Low Vision Taskforce provides details on their [Placeholder Research.](https://www.w3.org/WAI/GL/low-vision-a11y-tf/wiki/Placeholder_Research).

**Comment by aagonzales** [comment]
@snidersd those instructions aren't very clear to me. It doesn't explicitly say placeholder text must pass accessibility standards. I wish these guidelines were clearer.

**Comment by dakahn** [comment]
The way I understand it is that there are contrast requirements on placeholder text when it's the only labeling for the form item.

Though in general if your placeholder text conveys any important information (format or whatever) then that information should be accessible to everyone.

 I think we should default to the more readable placeholder contrast since form items without labels are something we permit and (anecdotally) a pretty popular design decision.
